sfge/src/main/java/com/salesforce/rules/UnusedMethodRule.java

```diff
@@ -243,188 +241,296 @@ private boolean parameterTypesMatch(
         return true;
     }
 
-    private Optional<BaseSFVertex> getTypeDeclaration(ChainedVertex chainedVertex) {
+    private String getDeclaredType(BaseSFVertex vertex) {
+        if (vertex instanceof Typeable) {
+            return ((Typeable) vertex).getCanonicalType();
+        } else if (vertex instanceof MethodVertex) {
+            return ((MethodVertex) vertex).getReturnType();
+        } else if (vertex instanceof UserClassVertex || vertex instanceof UserInterfaceVertex) {
+            // Can cast to NamedVertex since both classes implement NamedVertex.
+            return ((NamedVertex) vertex).getName();
+        } else {
+            return null;
+        }
+    }
+
+    /** Identify the vertex that represents a declaration of the thing being invoked. */
+    private Optional<BaseSFVertex> findDeclaration(ChainedVertex chainedVertex) {
```

Comment:
Since this code is still young, can we start using the visitor pattern for instanceof checks?
We've been trying to move out of the long if/else structures and Jeff had created [TypedVertexVisitor parent class](https://github.com/forcedotcom/sfdx-scanner/blob/dev-3/sfge/src/main/java/com/salesforce/graph/visitor/TypedVertexVisitor.java).

Example of how it can be used: https://github.com/forcedotcom/sfdx-scanner/blob/dev-3/sfge/src/main/java/com/salesforce/graph/ops/expander/switches/SObjectWhenBlockExcluder.java